actors/dqn.py

```diff
@@ -0,0 +1,524 @@
+        self._dtype = dtype
+        self._environment_implementation = environment_implementation
+        self._num_input = num_input
+        self._num_output = num_output
+        self._num_hidden_nodes = list(num_hidden_nodes)
+
+        self.epsilon = epsilon
+        self.base_network = MLPNetwork(num_input, self._num_hidden_nodes)
+        self.network = DQNNetwork(self.base_network, self._num_hidden_nodes[-1], self._num_output)
+
+        # version user data
+        self.num_samples_seen = 0
+
+    def get_model_user_data(self):
+        return {
+            "environment_implementation": self._environment_implementation,
+            "num_input": self._num_input,
+            "num_output": self._num_output,
+            "num_hidden_nodes": json.dumps(self._num_hidden_nodes),
+        }
+
+    def save(self, model_data_f):
+        torch.save((self.network.state_dict(), self.epsilon), model_data_f)
+
+        return {"num_samples_seen": self.num_samples_seen}
+
+    @staticmethod
+    def serialize_model(model) -> bytes:
+        stream = io.BytesIO()
+        torch.save(
+            (
+                model.network.state_dict(),
+                model.epsilon,
+                model.get_model_user_data(),
+            ),
+            stream,
+        )
+        return stream.getvalue()
+
+    @classmethod
+    def deserialize_model(cls, serialized_model) -> DQNModel:
+        stream = io.BytesIO(serialized_model)
+        (network_state_dict, epsilon, model_user_data) = torch.load(stream)
+
+        model = cls(
+            model_id=model_user_data["model_id"],
+            environment_implementation=model_user_data["environment_implementation"],
+            num_input=int(model_user_data["num_input"]),
+            num_output=int(model_user_data["num_output"]),
+            num_hidden_nodes=json.loads(model_user_data["num_hidden_nodes"]),
+            epsilon=0,
+        )
+        model.network.load_state_dict(network_state_dict)
+        model.epsilon = epsilon
+        model.num_samples_seen = int(model_user_data["num_samples_seen"])
+
+        return model
+
+
+class DQNActor:
+    def __init__(self, _cfg):
+        self._dtype = torch.float
+        self.samples_since_update = 0
+
+    def get_actor_classes(self):
+        return [PLAYER_ACTOR_CLASS]
+
+    async def impl(self, actor_session):
+        actor_session.start()
+
+        config = actor_session.config
+
+        rng = np.random.default_rng(config.seed if config.seed is not None else 0)
+
+        environment_specs = EnvironmentSpecs.deserialize(config.environment_specs)
+        observation_space = environment_specs.get_observation_space()
+        action_space = environment_specs.get_action_space(seed=rng.integers(9999))
+
+        assert isinstance(action_space.gym_space, Discrete)
+
+        model, _, _ = await actor_session.model_registry.retrieve_version(
+            DQNModel, config.model_id, config.model_version
+        )
```

Comment:
The word version was completely disassociated from the `model_registry_v2`. All instances of `version` or `model_version` are now `iteration` or `model_iteration`. This is the case for `config.model_iteration`.